app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -531,33 +535,35 @@ class NewMessageViewModel @Inject constructor(
         }
     }
 
-    private fun importAttachments(uris: List<Uri>, draft: Draft) = viewModelScope.launch(ioCoroutineContext) {
+    fun importAttachments(currentAttachments: List<Attachment>, uris: List<Uri>): List<Attachment> {
 
         val newAttachments = mutableListOf<Attachment>()
-        var attachmentsSize = draft.attachments.sumOf { it.size }
+        var attachmentsSize = currentAttachments.sumOf { it.size }
         var result = ImportationResult.SUCCESS
 
         uris.forEach { uri ->
             val availableSpace = ATTACHMENTS_MAX_SIZE - attachmentsSize
-            val (attachment, hasSizeLimitBeenReached) = importAttachment(draft.localUuid, uri, availableSpace) ?: return@forEach
+            val (attachment, hasSizeLimitBeenReached) = importAttachment(uri, availableSpace) ?: return@forEach
 
-            if (hasSizeLimitBeenReached) result = ImportationResult.FILE_SIZE_TOO_BIG
+            if (hasSizeLimitBeenReached) result = ImportationResult.ATTACHMENTS_TOO_BIG
 
             attachment?.let {
                 newAttachments.add(it)
                 attachmentsSize += it.size
             }
         }
 
-        importedAttachments.postValue(newAttachments to result)
+        importAttachmentsResult.postValue(result)
+
+        return newAttachments
     }
 
-    private fun importAttachment(localUuid: String, uri: Uri, availableSpace: Long): Pair<Attachment?, Boolean>? {
+    private fun importAttachment(uri: Uri, availableSpace: Long): Pair<Attachment?, Boolean>? {
 
         val (fileName, fileSize) = context.getFileNameAndSize(uri) ?: return null
         val attachment = Attachment()
 
-        return LocalStorageUtils.saveAttachmentToUploadDir(context, uri, fileName, localUuid, attachment.localUuid)
+        return LocalStorageUtils.saveAttachmentToUploadDir(context, uri, fileName, draftLocalUuid!!, attachment.localUuid)
```

Comment:
this can be called before initLiveData, so there is a risk that draftLocalUuid is null

Reply:
we should probably keep this ref to localUuid from the draft (or call the initLiveData sooner)

Reply:
This is actually really called before `initLiveData`, so there is currently a bug.
Nice catch !